app/src/main/java/com/example/fitjournal/journalEntry/JournalEntryScreen.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.fitjournal.journalEntry
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.example.fitjournal.core.presentation.commoncomponents.textField.SearchBar
+import com.example.fitjournal.core.presentation.navigation.NavigationInterface
+import com.example.fitjournal.core.presentation.theme.Spacing
+import com.example.fitjournal.journalEntry.components.JournalEntryList
+import com.example.fitjournal.journalEntry.domain.JournalEntryViewModel
+import com.example.fitjournal.journalEntry.domain.WorkoutDetail
+
+@Composable
+fun JournalEntryScreen(
+    modifier: Modifier,
+    selectedJournalEntry: (WorkoutDetail) -> Unit,
+    navigateToDestination: (NavigationInterface) -> Unit
+) {
+    val viewModel: JournalEntryViewModel = viewModel()
```

Comment:
Sure, just curious it is that normal Android practice for the main component to own all the ViewModels?

Reply:
In compose approach, yes since the main activity is the sole activity that persists on launch and composables are just rendered ui functional components